### DISCUSSION TITLE
Update column in related table with values from own table

### DISCUSSION THREAD

**Discussion by NixBiks** [prisma/prisma#8307] [Q&A] [answered]
I have `product` table and `productAudit` table and they are related through a `productId`.

```prisma
model Product {
  createdAt       DateTime     @default(now())
  id Int @id @default(autoincrement())
}

model ProductAudit {
  timestamp       DateTime     @default(now())
  productId Int
  product   Product @relation(fields: [productId], references: [id])
  @@unique([productId, timestamp])
}
```

I want to go through all products and update all audit timestamps with create at timestamp from product. The following does the trick but it seems unnecessarily heavy since I read all audits, then for each audit I loop up the related product. I use that to update the current audit. So 2 reads plus 1 write for each row in audit.

```typescript
// update audit timestamps
for (const product of await prisma.productAudit.findMany()) {
  const timestamp = (await prisma.product.findUnique({
    where: { id: product.productId },
  }))!.createdAt;
  await prisma.productAudit.update({
    data: { timestamp },
    where: {
      productId_timestamp: {
        productId: product.productId,
        timestamp: product.timestamp,
      },
    },
  });
}
```

**Comment by ryands17** [comment]
@mr-bjerre 👋 

I can't think of anything here to optimise this unfortunately. @2color wdyt?

**Comment by 2color** [comment, marked as answer]
Hey @mr-bjerre 👋,

Which database are you using? 
Is this a one-off update that you want to do?


As you mentioned this approach is a bit heavy as it requires a read and a write for each row in the audit table in addition to the `findMany` call.

It's possible to do this in a single query using raw SQL and there should be a feature request open about allowing updating values in Prisma Client operations based on the value of another column.

With the current approach you're taking, my advice to speed up this operation would be to fetch the related product in the `findMany` call and encapsulate all the `update` operations in a transaction as follows:

```ts
const updateTxs = []
  for (const audit of await prisma.productAudit.findMany({
    include: { product: true },
  })) {
    const timestamp = audit.product.createdAt
    updateTxs.push(
      prisma.productAudit.update({
        data: { timestamp },
        where: {
          productId_timestamp: {
            productId: audit.productId,
            timestamp: audit.timestamp,
          },
        },
      }),
    )
  }

  await prisma.$transaction(updateTxs)
```

This should make the script much faster to run because it doesn't need to wait for every single update operation before moving to the next.

**Reply by NixBiks** [reply]
I did not know about `prisma.$transaction`. Brilliant, thanks !

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
